lego/apps/surveys/serializers.py

```diff
@@ -0,0 +1,160 @@
+        fields = ('id', 'user', 'answers')
+
+    def create(self, validated_data):
+        survey = Survey.objects.get(pk=self.context['view'].kwargs['survey_pk'])
+        answers = None if 'answers' not in validated_data else validated_data.pop('answers')
+        submission = Submission.objects.create(survey=survey, **validated_data)
+        submission.save()
+
+        if answers is not None:
+            for answer in answers:
+                question = answer.pop('question')
+
+                if getattr(question, 'question_type') in [1, 2]:
+                    if getattr(question, 'question_type') is 1 and \
+                                    len(answer['selected_options']) > 1:
+                        raise exceptions.ValidationError('You cannot select multiple options for '
+                                                         'this type of question.')
+
+                    Answer.create(submission, question, **answer)
+                else:
+                    new_answer = Answer.objects.create(submission=submission, question=question,
+                                                       **answer)
+                    new_answer.save()
+
+        return submission
+
+
+class SurveyReadDetailedSerializer(BasisModelSerializer):
+    questions = QuestionSerializer(many=True)
+    event = EventReadSerializer()
+
+    class Meta:
+        model = Survey
+        fields = ('id', 'title', 'active_from', 'questions', 'event', 'template_type')
+
+
+class SurveyCreateSerializer(BasisModelSerializer):
+    questions = QuestionSerializer(many=True, required=False, allow_null=True)
+
+    class Meta:
+        model = Survey
+        fields = ('id', 'title', 'active_from', 'template_type', 'event', 'questions')
+
+    def create(self, validated_data):
+        questions = validated_data.pop('questions')
+        survey = Survey.objects.create(**validated_data)
+
+        for question in questions:
+            options = question.pop('options') if 'options' in question else None
+            new_question = Question.objects.create(survey=survey, **question)
+
+            if options is not None:
+                for option in options:
+                    Option.objects.create(question=new_question, **option)
+
+        return survey
+
+
+class SurveyUpdateSerializer(BasisModelSerializer):
+    questions = QuestionUpdateSerializer(many=True, required=False, allow_null=True)
+
+    class Meta:
+        model = Survey
+        fields = ('id', 'title', 'active_from', 'template_type', 'event', 'questions')
+
+    def update(self, instance, validated_data):
+        questions = validated_data.pop('questions')
+        for key, value in validated_data.items():
+            setattr(instance, key, value)
+        instance.save()
+
+        for question in questions:
+            options = question.pop('options') if 'options' in question else None
+            Question.objects.filter(id=question['id']).update(**question)
```

Comment:
What happens if the filter returns None? Get instead of filter?

Reply:
https://docs.djangoproject.com/en/1.9/topics/db/queries/#updating-multiple-objects-at-once
The `update` function is defined to work on a QuerySet, not a single object. The "appropriate" way to do it would be something like
```
q = Question.objects.get(...)
q.someValue = ...
q.otherValue = ...
q.save()
```

But that seem tedious, and [this StackOverflow post](https://stackoverflow.com/questions/2712682/django-update-object) seems to indicate that `update` is fine (or even better) for single objects too.

Reply:
:+1: